include/aikido/robot/ros/RosRobot.hpp

```diff
@@ -18,21 +21,198 @@ AIKIDO_DECLARE_POINTERS(RosRobot)
+  /// \return True if successful. If false, all executors are inactive.
+  bool activateExecutor(std::string id) override; 
+
+  using Robot::activateExecutor; // This un-hides Robot::activateExecutor(const aikido::control::ExecutorType type)
+
+  ///
+  /// Sets the ros controller service client which enables controller loading.
+  ///
+  /// \param[in] rosControllerServiceClient ros controller service client.
+  void setRosLoadControllerServiceClient(
+      const std::shared_ptr<::ros::ServiceClient>& rosLoadControllerServiceClient); 
+
+  ///
+  /// Sets the ros controller service client which enables controller switching.
+  ///
+  /// \param[in] rosControllerServiceClient ros controller service client.
+  void setRosSwitchControllerServiceClient(
+      const std::shared_ptr<::ros::ServiceClient>& rosSwitchControllerServiceClient);
+
+  ///
+  /// Sets the ros joint mode command client which enables controller mode switching.
+  ///
+  /// \param[in] rosJointModeCommandClient ros joint mode command client.
+  void setRosJointModeCommandClient(
+      const std::shared_ptr<aikido::control::ros::RosJointModeCommandClient>& rosJointModeCommandClient);
+
+  ///
+  /// Starts the controller with name startControllerName.
+  ///
+  /// \param[in] startControllerName name of controller to start.
+  /// \return True if successful. Otherwise false.
+  bool startController(const std::string startControllerName);
+
+  ///
+  /// Stops the controller with name stopControllerName.
+  ///
+  /// \param[in] stopControllerName name of controller to stop.
+  /// \return True if successful. Otherwise false.
+  bool stopController(const std::string stopControllerName);
+
+  ///
+  /// Stops the controller with name stopControllerName and starts the
+  /// controller with name startControllerName.
+  ///
+  /// \param[in] startControllersNames list of names of controllers to start.
+  /// \param[in] stopControllersNames list of names of controllers to stop.
+  /// \return True if successful. Otherwise false.
+  bool switchControllers(
+      const std::vector<std::string> startControllersNames, 
+      const std::vector<std::string> stopControllersNames);
+
+  ///
+  /// Loads the controller with name loadControllerName.
+  ///
+  /// \param[in] loadControllerName name of controller to load.
+  /// \return True if successful. Otherwise false.
+  bool loadController(const std::string loadControllerName);
+
+  ///
+  /// Switches the controller mode to jointMode.
+  ///
+  /// \param[in] jointMode control mode to switch to.
+  /// \return True if successful. Otherwise false.
+  bool switchControllerMode(
+      const hardware_interface::JointCommandModes jointMode);
+
+protected:
+  std::unordered_map<std::string, hardware_interface::JointCommandModes> mRosControllerModes;
+  std::unordered_map<std::string, std::string> mRosControllerNames;
+
+  // Ros node associated with this robot.
+  std::shared_ptr<::ros::NodeHandle> mNode;
```

Comment:
As in other classes, don't have this be a shared pointer. Have it be a non-pointer object, and just initialize it in the constructor using the passed in object (if provided). If not provided, create it manually.